csaf-lib/src/csaf/csaf2_1/validation.rs

```diff
@@ -111,4 +116,27 @@ mod tests {
             Err(String::from("Recursion depth too big: 31"))
         )
     }
+
+    #[test]
+    fn test_test_6_01_35() {
+        for x in ["11", "12", "13", "14"].iter() {
+            let doc = load_document(format!("../csaf/csaf_2.1/test/validator/data/mandatory/oasis_csaf_tc-csaf_2_1-2024-6-1-35-{}.json", x).as_str()).unwrap();
+            assert_eq!(
+                Ok(()),
+                test_6_01_35_contradicting_remediations(&doc)
+            )
+        }
+        for (x, err) in [
+            ("01", "Product CSAFPID-9080700 has contradicting remediations: no_fix_planned and vendor_fix"),
+            ("02", "Product CSAFPID-9080700 has contradicting remediations: none_available and mitigation"),
+            ("03", "Product CSAFPID-9080702 has contradicting remediations: workaround, fix_planned and optional_patch"),
+            ("04", "Product CSAFPID-9080701 has contradicting remediations: mitigation, fix_planned and optional_patch"),
+        ].iter() {
+            let doc = load_document(format!("../csaf/csaf_2.1/test/validator/data/mandatory/oasis_csaf_tc-csaf_2_1-2024-6-1-35-{}.json", x).as_str()).unwrap();
+            assert_eq!(
+                Err(format!("{}", err)),
+                test_6_01_35_contradicting_remediations(&doc)
+            )
+        }
+    }
```

Comment:
Can we have one output line per tested file? This makes it easier to spot if a file is missed.

Reply:
Sure. Will implement an output of the file under test.

Reply:
Testing output is already in place, it turns out.
`cargo test -- --nocapture` will output information about parsed docs.
If the scattered output (due to multiple test threads) is an issue: `cargo test -- --nocapture --test-threads 1`